### PR TITLE
Bug 2090829: Bump OpenShift router to k8s 1.24 and go 1.18

### PR DESCRIPTION
**Pull request by candita**
Update vendor packages to v0.24.1 for:

k8s.io/api
k8s.io/apimachinery
k8s.io/apiserver
k8s.io/client-go

Update router code for changes to vendor packages:

pkg/router/controller/router_controller.go
pkg/router/controller/status_test.go

### CODE REVIEW

**Comment by candita** [comment]
The e2e-upgrade test failure is likely due to a configuration issue in CI. I asked for permission to override in 
https://coreos.slack.com/archives/C01CQA76KMX/p1654121250686089

**Comment by candita** [comment]
> failed: (17.2s) 2022-05-27T22:40:25 "[sig-auth][Feature:OAuthServer] [Token Expiration] Using a OAuth client with a non-default token max age to generate tokens that do not expire works as expected when using a code authorization flow [Suite:openshift/conformance/parallel]"
/test e2e-metal-ipi-ovn-ipv6

File: pkg/router/controller/router_controller.go [hunk 1]
```diff
@@ -238,7 +238,6 @@ func (c *RouterController) HandleEndpointSlice(eventType watch.EventType, objMet
 			Labels:          objMeta.Labels,
 			Annotations:     objMeta.Annotations,
 			OwnerReferences: objMeta.OwnerReferences,
-			ClusterName:     objMeta.ClusterName,
```
Comment (gcs278) [inline pkg/router/controller/router_controller.go:241]: Just curious, where did this change come from?

File: pkg/router/controller/status_test.go [hunk 2]
```diff
@@ -896,3 +896,7 @@ func (i *fakeInformer) HasSynced() bool {
 func (i *fakeInformer) LastSyncResourceVersion() string {
 	panic("not implemented")
 }
+
+func (i *fakeInformer) SetTransform(handler cache.TransformFunc) error {
```
Comment (gcs278) [inline pkg/router/controller/status_test.go:900]: And this change as well - why it is required for the bump?

File: pkg/router/controller/router_controller.go (hunk 1 above)
Comment (candita) [reply pkg/router/controller/router_controller.go:241]: The field was deprecated in the vendor bump and not used anywhere.  I'll dig up the exact source.

Comment (candita) [reply pkg/router/controller/router_controller.go:241]: https://github.com/kubernetes/apimachinery/commit/430b920312ca0fa10eca95967764ff08f34083a3

Comment (gcs278) [reply pkg/router/controller/router_controller.go:241]: Ah interesting thanks!

File: pkg/router/controller/status_test.go (hunk 2 above)
Comment (candita) [reply pkg/router/controller/status_test.go:900]: https://github.com/kubernetes/client-go/commit/54928eef9f824667b23a938188498992d437156a

**Comment by gcs278** [comment]
Seems quite straight forward to me. LGTM but I'll let someone else approve.

**Comment by candita** [comment]
https://github.com/openshift/origin/pull/27181 has merged

/retest

**Comment by candita** [comment]
We may also need to wait for https://github.com/openshift/origin/pull/27229 to merge.

**Comment by candita** [comment]
> Cluster operator etcd Degraded is True with UpgradeBackupController_Error: UpgradeBackupControllerDegraded: unable to retrieve cluster version, no completed update was found in cluster version status history: [{Partial 2022-06-08 17:03:20 +0000 UTC <nil> 4.11.0-0.ci.test-2022-06-08-164802-ci-op-j349vdt3-latest registry.build01.ci.openshift.org/ci-op-j349vdt3/release@sha256:5c88274a74f2b2eb2503e361e9790220859b232d505d08909cc7570b348f2a20 false }]

**Comment by candita** [comment]
e2e-agnostic has a known issue with azure:
> Cluster operator etcd Degraded is True with UpgradeBackupController_Error: UpgradeBackupControllerDegraded: unable to retrieve cluster version, no completed update was found in cluster version status history: [{Partial 2022-06-08 22:21:41 +0000 UTC <nil> 4.11.0-0.ci.test-2022-06-08-220558-ci-op-j349vdt3-latest registry.build01.ci.openshift.org/ci-op-j349vdt3/release@sha256:bcadfa193f01fb1ef344d87dc6721752336d6a1167b91c3334d40e14e81ed159 false }]

https://coreos.slack.com/archives/C01CQA76KMX/p1654734729402019

**Comment by candita** [comment]
error msg=Cluster operator etcd Degraded is True with UpgradeBackupController_Error: UpgradeBackupControllerDegraded: unable to retrieve cluster version, no completed update was found in cluster version status history: [{Partial 2022-06-09 17:01:49 +0000 UTC <nil> 4.11.0-0.ci.test-2022-06-09-164534-ci-op-j1mw735x-latest registry.build01.ci.openshift.org/ci-op-j1mw735x/release@sha256:515bb0a39c23773031104a64206fcf9a104d745aaf19d5c62f345126afe18b5a false }]

level=info msg=Cluster operator insights SCAAvailable is False with NotFound: Failed to pull SCA certs from https://api.openshift.com/api/accounts_mgmt/v1/certificates: OCM API https://api.openshift.com/api/accounts_mgmt/v1/certificates returned HTTP 404: {"code":"ACCT-MGMT-7","href":"/api/accounts_mgmt/v1/errors/7","id":"7","kind":"Error","operation_id":"8886f88b-06b2-4fe9-9d72-40fc512fe435","reason":"The organization (id= 1UtjtUH8sxFO38rF9fAoZCkxMsN) does not have any certificate of type sca. Enable SCA at https://access.redhat.com/management."}

**Comment by candita** [comment]
> fail [github.com/openshift/origin/test/extended/util/disruption/backend_sampler_tester.go:182]: Jun 10 23:52:04.329: disruption/kube-api connection/new was unreachable during disruption: (no exact match for historicaldata.DataKey{Name:"kube-api-new-connections", JobType:platformidentification.JobType{Release:"4.11", FromRelease:"4.11", Platform:"azure", Architecture:"amd64", Network:"sdn", Topology:"ha"}}, fell back to historicaldata.DataKey{Name:"kube-api-new-connections", JobType:platformidentification.JobType{Release:"4.11", FromRelease:"4.10", Platform:"azure", Architecture:"amd64", Network:"sdn", Topology:"ha"}}) for at least 1m58s of 1h11m36s (maxAllowed=1m22s):

/test e2e-upgrade

**Comment by candita** [comment]
> fail [github.com/openshift/origin/test/extended/util/disruption/backend_sampler_tester.go:182]: Jun 10 23:52:04.329: disruption/kube-api connection/new was unreachable during disruption: (no exact match for historicaldata.DataKey{Name:"kube-api-new-connections", JobType:platformidentification.JobType{Release:"4.11", FromRelease:"4.11", Platform:"azure", Architecture:"amd64", Network:"sdn", Topology:"ha"}}, fell back to historicaldata.DataKey{Name:"kube-api-new-connections", JobType:platformidentification.JobType{Release:"4.11", FromRelease:"4.10", Platform:"azure", Architecture:"amd64", Network:"sdn", Topology:"ha"}}) for at least 1m58s of 1h11m36s (maxAllowed=1m22s):

/test e2e-upgrade

**Comment by candita** [comment]
> Alerting rule "MachineConfigControllerPausedPoolKubeletCA" (group: mcc-paused-pool-kubelet-ca) WARNING: Alert "MachineConfigControllerPausedPoolKubeletCA" has an invalid 'runbook_url' annotation: timed out waiting for the condition
/test e2e-aws-serial

**Pull request merged**